### PR TITLE
adding Negative Acknowledge to consumer

### PR DESCRIPTION
**Pull request by usaguerrilla**
# Summary

(Please provide a high level summary)

### CODE REVIEW

File: src/DotPulsar/Internal/Abstractions/IMessageTracker.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace DotPulsar.Internal.Abstractions
+{
+    public interface IMessageTracker
```
Comment (Chickenzilla) [inline src/DotPulsar/Internal/Abstractions/IMessageTracker.cs]: Was this obsoleted by a design change?  Looks unused.

Comment (usaguerrilla) [reply src/DotPulsar/Internal/Abstractions/IMessageTracker.cs]: good catch

File: src/DotPulsar/Internal/ConsumerChannel.cs [hunk 2]
```diff
@@ -155,6 +174,7 @@ public async ValueTask DisposeAsync()
         {
             try
             {
+                _cancellationTokenSource.Dispose();
```
Comment (Chickenzilla) [inline src/DotPulsar/Internal/ConsumerChannel.cs:178]: Umm, did you mean .Cancel()?  I don't see any linked token sources or registrations that would require disposal, and disposal doesn't seem to actually cancel the tokens either, so your loop stays running forever.

Comment (usaguerrilla) [reply src/DotPulsar/Internal/ConsumerChannel.cs:178]: good catch! I usually add Dispose anyway, as I got bitten several times by not adding it and then later adding functionality which requires it...

**Review by Chickenzilla — APPROVED** [review]
I believe this looks OK now, though I am mildly concerned by having an extra task running per `ConsumerChannel`.  I could envision a use case for a single process to have 100, perhaps even 1000 consumers open (multiple partitioned topics, etc).  We can maybe optimize this later by refactoring the timer check into one task at the client level that calls the process manager or something.  Since we only check for nack readiness every delay interval anyhow, we could also simplify the logic into two `ConcurrentDictionary`'s, a "current" and a "next", where new entries go into next, and current and next flip once you send everything in current.

**Comment by usaguerrilla** [comment]
yeh was thinking about this but will have to manage channels and time when channel dies.

though:
- amount of work to be done is the same (send ids to N channels)
- we do have await so tasks will get created anyway every time we send data
- tasks aren't really threads, and I do not think even 1000 of them is a big deal (only memory consumption is a mild issue)
- one uber-dictionary is a single choke point

will test with 5'000 (max in QA).

**Pull request merged**